code/modules/food_and_drinks/kitchen_machinery/kitchen_machine.dm

```diff
@@ -274,39 +257,81 @@
 			muck_finish()
 			fail()
 			return
-		else if(has_extra_item())
+		else if(has_extra_item())	//if extra items present, break down and require repair before next use
 			if(!wzhzhzh(4))
 				abort()
 				return
 			broke()
 			fail()
 			return
-		else
+		else	//otherwise just stop without requiring cleaning/repair
 			if(!wzhzhzh(10))
 				abort()
 				return
 			stop()
 			fail()
 			return
 	else
-		var/halftime = round(recipe.time/10/2)
-		if(!wzhzhzh(halftime))
+		if(!wzhzhzh(5))
 			abort()
 			return
-		if(!wzhzhzh(halftime))
+		if(!wzhzhzh(5))
 			abort()
 			fail()
 			return
-		cooked = recipe.make_food(src)
-		byproduct = recipe.get_byproduct()
-		stop()
-		if(cooked)
-			cooked.forceMove(loc)
-		for(var/i=1,i<efficiency,i++)
-			cooked = new cooked.type(loc)
-		if(byproduct)
-			new byproduct(loc)
+		make_recipes(recipes_to_make)
+
+/obj/machinery/kitchen_machine/proc/choose_recipes()	//picks out recipes for the machine and any mixing bowls it may contain, and builds a list of the selected recipes to be made in a later proc by associating the "source" of the ingredients (mixing bowl, machine) with the recipe for that source
```

Comment:
Please put the comments on separate lines before the clauses they're tied to, they can't be read like this